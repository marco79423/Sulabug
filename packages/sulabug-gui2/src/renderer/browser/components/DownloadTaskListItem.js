import * as React from 'react'
import {createStyles, withStyles} from '@material-ui/core/styles'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemAvatar from '@material-ui/core/ListItemAvatar'
import LinearProgress from '@material-ui/core/LinearProgress'
import Avatar from '@material-ui/core/Avatar'
import {useDispatch, useSelector} from 'react-redux'
import * as ducks from '../ducks'

const styles = (theme) => createStyles({
  root: {},
  progress: {
    marginRight: theme.spacing(2)
  }
})

function DownloadTaskListItem ({classes, downloadTaskId}) {
  const dispatch = useDispatch()
  const downloadTaskMap = useSelector(ducks.getDownloadTaskMap)
  const downloadTask = downloadTaskMap[downloadTaskId]


  const renderProgress = () => {
    switch (downloadTask.status) {
      case 'PREPARING':
        return <LinearProgress className={classes.progress} variant="query"/>
      case 'DOWNLOADING':
      case 'FINISHED':
        return <LinearProgress className={classes.progress} variant="determinate" value={downloadTask.progress}/>
      default:
        return null
    }
  }

  return (
    <ListItem className={classes.root}>
      <ListItemAvatar>
        <Avatar>
          <img src={downloadTask.coverUrl}/>
        </Avatar>
      </ListItemAvatar>
      <ListItemText
        primary={downloadTask.name}
        secondary={renderProgress()}
      />
    </ListItem>
  )
}

export default withStyles(styles)(DownloadTaskListItem)
