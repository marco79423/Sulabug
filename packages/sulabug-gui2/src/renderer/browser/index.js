import React, {useEffect} from 'react'
import {useDispatch} from 'react-redux'
import {Redirect, Route, Switch} from 'react-router-dom'
import {createMuiTheme, ThemeProvider} from '@material-ui/core/styles'
import {lime, teal} from '@material-ui/core/colors'

import CollectionPage from './pages/CollectionPage'
import BrowsePage from './pages/BrowsePage'
import SettingsPage from './pages/SettingsPage'
import DownloadPage from './pages/DownloadPage'


export default function Browser() {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch({type: 'common/initialize-app', data: 'browser'})
  }, [])

  const theme = createMuiTheme({
    typography: {
      h2: {
        fontSize: '1.6rem',
      },
      subtitle2: {
        fontSize: '0.8rem',
      },
      body2: {
        fontSize: '0.9rem',
      }
    },
    palette: {
      primary: {
        main: teal[800]
      },
      secondary: {
        main: lime[600]
      },
    },
  })

  return (
    <ThemeProvider theme={theme}>
      <Switch>
        <Redirect exact from='/browser' to='/browser/collection'/>
        <Route exact path='/browser/collection' component={CollectionPage}/>
        <Route exact path='/browser/browse' component={BrowsePage}/>
        <Route exact path='/browser/download' component={DownloadPage}/>
        <Route exact path='/browser/settings' component={SettingsPage}/>
      </Switch>
    </ThemeProvider>
  )
}
