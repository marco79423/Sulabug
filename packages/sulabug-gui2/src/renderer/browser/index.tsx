import React, {useEffect, useReducer} from 'react'
import {ipcRenderer} from 'electron'
import {Link} from 'react-router-dom'

console.log('browser module loaded')

function reducer(state, action) {
  switch (action.type) {
    case 'add-log':
      return {...state, logs: [...state.logs, action.data]}
    default:
      return state
  }
}

export default function Browser() {
  const [state, dispatch] = useReducer(reducer, {logs: []})
  useEffect(() => {
    ipcRenderer.on('sulabug-action', (e, action) => {
      dispatch(action)
    })
  }, [])

  return (
    <div>
      browser <Link to={`/manager`}>manager</Link>
      <ul>
        {state.logs.map((log, i) => <li key={i}>{log}</li>)}
      </ul>
    </div>
  )
}
