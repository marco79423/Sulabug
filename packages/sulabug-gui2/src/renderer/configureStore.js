import {applyMiddleware, combineReducers, compose, createStore} from 'redux'
import {forwardToMain, replayActionRenderer} from 'electron-redux'
import createSagaMiddleware from 'redux-saga'

import managerReducer from './manager/reducer'
import browserReducer from './browser/reducer'
import rootSaga from './rootSaga'


export default function configureStore() {
  const reducer = combineReducers({
    manager: managerReducer,
    browser: browserReducer,
  })

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose

  const sagaMiddleware = createSagaMiddleware()

  const store = createStore(
    reducer,
    composeEnhancers(applyMiddleware(
      forwardToMain, // IMPORTANT! This goes first
      sagaMiddleware,
    ))
  )

  sagaMiddleware.run(rootSaga)

  replayActionRenderer(store)

  return store
}
