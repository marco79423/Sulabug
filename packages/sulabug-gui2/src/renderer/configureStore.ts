import {applyMiddleware, combineReducers, compose, createStore} from 'redux'
import managerReducer from './manager/reducer'
import browserReducer from './browser/reducer'

declare global {
  interface Window { __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: any; }
}


export default function configureStore() {
  const reducer = combineReducers({
    manager: managerReducer,
    browser: browserReducer,
  })

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose

  const store = createStore(
    reducer,
    composeEnhancers(applyMiddleware(
    ))
  )

  return store
}
