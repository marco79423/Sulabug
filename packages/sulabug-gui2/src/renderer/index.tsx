import React, {lazy, Suspense} from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'

const Manager = lazy(() => import('./manager'))
const Browser = lazy(() => import('./browser'))

function App() {
  return (
    <HashRouter>
      <Suspense fallback={<div>loading</div>}>
        <Switch>
          <Redirect exact from='/' to='/manager'/>
          <Route path="/manager" component={Manager}/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Suspense>
    </HashRouter>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
