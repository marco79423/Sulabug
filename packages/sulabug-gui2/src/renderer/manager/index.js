import React, {useEffect} from 'react'
import {ipcRenderer} from 'electron'
import {useDispatch, useSelector} from 'react-redux'
import {Link} from 'react-router-dom'


export default function Manager() {
  const dispatch = useDispatch()
  const count = useSelector(state => state.manager.count)

  useEffect(() => {
    dispatch({type: 'common/initialize-app', data: 'manager'})
  }, [])


  useEffect(() => {
    setTimeout(() => {
      console.log('count', count)
      ipcRenderer.send('sulabug-action', {type: 'browser/add-log', data: count})
      dispatch({type: 'manager/increment-count'})
    }, 1000)
  }, [count])

  return (
    <div>
      manager <Link to='/browser'>browser</Link>
    </div>
  )
}
