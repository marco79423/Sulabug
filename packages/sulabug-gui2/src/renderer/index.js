import React from 'react'
import * as ReactDOM from 'react-dom'
import {HashRouter as Router, Route, Switch} from 'react-router-dom'
import {Redirect} from 'react-router'
import {Provider} from 'react-redux'

import configureStore from './configureStore'
import Browser from './browser'

const store = configureStore()

function App() {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Redirect exact from='/' to='/browser'/>
          <Route path="/browser" component={Browser}/>
        </Switch>
      </Router>
    </Provider>
  )
}

ReactDOM.render(
  <App/>,
  document.getElementById('app'),
)
