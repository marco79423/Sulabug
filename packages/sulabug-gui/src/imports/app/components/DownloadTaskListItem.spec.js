import 'reflect-metadata'

import React from 'react'
import renderer from 'react-test-renderer'

import DownloadTaskListItem from './DownloadTaskListItem'


describe('DownloadTaskListItem', () => {
  it('will render download task correctly', async () => {
    const component = renderer.create(
      <DownloadTaskListItem
        downloadTask={{
          id: 'id',
          name: 'name',
          coverDataUrl: 'coverDataUrl',
          sourceUrl: 'sourceUrl',
          status: 'status',
          progress: 10,
        }}
        deleteDownloadTask={() => {
        }}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
