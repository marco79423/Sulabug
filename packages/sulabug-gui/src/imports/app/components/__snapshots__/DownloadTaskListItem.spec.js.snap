// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadTaskListItem will render download task correctly 1`] = `
<li
  className="MuiListItem-root-3 MuiListItem-default-6 MuiListItem-dense-7 MuiListItem-gutters-11 DownloadTaskListItem-root-1"
  disabled={false}
>
  <div
    className="MuiAvatar-root-18 MuiAvatar-colorDefault-19"
  >
    <img
      src="coverDataUrl"
    />
  </div>
  <div
    className="MuiListItemText-root-21"
  >
    <span
      className="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
    >
      name
    </span>
  </div>
</li>
`;
